### DISCUSSION TITLE
Pole figure 3D presentation

### DISCUSSION THREAD

**Discussion by m-amrei** [mtex-toolbox/mtex#1955] [Tips and Tricks]
![PoleFigure3D_PresentationOfAnyHKL](https://github.com/mtex-toolbox/mtex/assets/59927140/c6165b9f-2675-4748-a504-a786f34ec769)
Hello, 
I wrote the following function in order to present a random set of data points hkl on a 3D sphere surface.
It has been proven to be practical in my case. I would be happy if that could help others (feel free to share if you find it useful, no credit required).

   `
    function plotPF3D(ebsd, hkl, numPoints, colorMap)
    % Function to plot 3D pole figure for a given hkl index and number of points

    % Input:
    % ebsd: ebsd variable containing the orientation data
    % hkl: 1x3 vector for the Miller indices (e.g., [1 0 0])
    % numPoints: Number of points to be sampled for plotting
    % colorMap: color map name
 

    % Check if the number of ebsd data points is less than numPoints
    if length(ebsd) <= numPoints
        ebsd_sample = ebsd;
    else
        idx = randperm(length(ebsd), numPoints);  % get random indices
        ebsd_sample = ebsd(idx);
    end

    o_sample = ebsd_sample.orientations;

    % Create a Miller object for the given hkl index
    h = Miller(hkl(1), hkl(2), hkl(3), ebsd.CS);
    h_sym = symmetrise(h); % thinking of symmetries

    directions_hkl_sym = o_sample * h_sym;

    % Convert matrices to vectors
    x = directions_hkl_sym.x(:);
    y = directions_hkl_sym.y(:);
    z = directions_hkl_sym.z(:);

    % Visualization
    figure;
    scatter3(x,y,z,1,abs(z), 'filled');
    axis equal;
   
    % titles or no titles? 
    title(['3D Pole Figure of (', num2str(hkl), ') planes for ', num2str(numPoints/1000),'k random points']);
    xticks([]);
    yticks([]);
    zticks([]);
   
    xlabel('X');
    ylabel('Y');
    zlabel('Z');
    set(gcf, 'Color', 'white'); % Set figure background color
    colormap(colorMap); % Customize the colormap
    colorbar('Ticks', [0, 1], 'TickLabels', {'-Z', '+Z'}); % Add color bar with labels
end`

**Comment by kilir** [comment]
Hi,
thank you for sharing this. You may also want to have a look at the '3d' option for vector3d/plot() or orientation/plotPDF() e.g.

```
cs = crystalSymmetry('422')
ori = orientation.rand(1000,cs)
h = Miller({1,0,0},{0,0,1},{1,1,2},cs)
plotPDF(ori,ori.phi1,h,'MarkerSize',3,'3d')
```

Cheers,
Rüdiger
